src/ios/PushPlugin.m

```diff
@@ -353,35 +362,36 @@ - (UNNotificationAction *)createAction:(NSDictionary *)dictionary {
 
 - (void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
     if (self.callbackId == nil) {
-        NSLog(@"Unexpected call to didRegisterForRemoteNotificationsWithDeviceToken, ignoring: %@", deviceToken);
+        NSLog(@"[PushPlugin] Unexpected call to didRegisterForRemoteNotificationsWithDeviceToken");
         return;
     }
-    NSLog(@"Push Plugin register success: %@", deviceToken);
 
-#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
-    // [deviceToken description] is like "{length = 32, bytes = 0xd3d997af 967d1f43 b405374a 13394d2f ... 28f10282 14af515f }"
-    NSString *token = [self hexadecimalStringFromData:deviceToken];
-#else
-    // [deviceToken description] is like "<124686a5 556a72ca d808f572 00c323b9 3eff9285 92445590 3225757d b83967be>"
-    NSString *token = [[[[deviceToken description] stringByReplacingOccurrencesOfString:@"<"withString:@""]
-                        stringByReplacingOccurrencesOfString:@">" withString:@""]
-                       stringByReplacingOccurrencesOfString: @" " withString: @""];
-#endif
+    NSLog(@"[PushPlugin] didRegisterForRemoteNotificationsWithDeviceToken");
 
-#if !TARGET_IPHONE_SIMULATOR
+    if([self usesFCM]) {
+        NSLog(@"[PushPlugin] Setting APNS Token to FCM");
+        [[FIRMessaging messaging] setAPNSToken:deviceToken];
+        [self initFirebaseMessaging:[self iOSOptions]];
+    }
 
     // Check what Notifications the user has turned on.  We registered for all three, but they may have manually disabled some or all of them.
-
-    UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
-    __weak PushPlugin *weakSelf = self;
-    [center getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings * _Nonnull settings) {
-
-        if(![weakSelf usesFCM]) {
-            [weakSelf registerWithToken: token];
+    [[UNUserNotificationCenter currentNotificationCenter] getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings * _Nonnull settings) {
+        if(![self usesFCM]) {
```

Comment:
`getNotificationSettingsWithCompletionHandler` is not being used anymore, it used to have code to check the settings, but they are not being checked anymore, so can be removed